core_impl/src/test/java/com/google/instrumentation/trace/SpanImplTest.java

```diff
@@ -63,15 +70,47 @@ public void toSpanData_NoRecordEvents() {
             SPAN_NAME,
             parentSpanId,
             false,
+            TraceParams.DEFAULT,
             startEndHandler,
             timestampConverter,
             testClock);
+    span.addAnnotation(Annotation.fromDescription(ANNOTATION_DESCRIPTION));
+    span.addAnnotation(ANNOTATION_DESCRIPTION, attributes);
+    span.addNetworkEvent(NetworkEvent.builder(NetworkEvent.Type.RECV, 1).setMessageSize(3).build());
+    span.addLink(Link.fromSpanContext(spanContext, Link.Type.CHILD));
```

Comment:
Why does this test need to add annotations, network events, and links?  If you are checking that those method calls don't throw exceptions when recording is off, I think it would help to have a comment and/or split it into a separate test.